src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -71,31 +71,34 @@ private int getHashcode(K key) {
         return key.hashCode();
     }
 
-    private boolean doPut(K key, V value) {
+    private void doPut(K key, V value) {
         Node<K, V> newNode = new Node<>(null, key, value);
         int index = getIndex(key);
         Node<K, V> currentNode = buckets[index];
 
         if (currentNode == null) {
             buckets[index] = newNode;
+            size++;
         } else {
-            while (true) {
-                if (checkNode(currentNode, key)) {
+            while (currentNode != null) {
```

Comment:
this is equivalent to `while(true)`: we are getting there only if `currentNode != null` (lines 79-81), we reassign it on line 94 with `currentNode.next`, but there is no way it will be null since in that case, we would return on line 92. Please make this code more readable by setting the while condition to `currentNode.next != null`, then you won't need code on lines 89-93, only `currentNode.next = newNode;` just after the while loop